zfmt/dag.go

```diff
@@ -130,6 +131,44 @@ func (c *canonDAG) expr(e dag.Expr, parent string) {
 		c.write("type<")
 		c.typ(e.Value)
 		c.write(">")
+	case *dag.RecordExpr:
+		c.write("{")
+		for k, elem := range e.Elems {
+			if k != 0 {
+				c.write(",")
+			}
+			switch e := elem.(type) {
+			case *dag.Field:
+				c.write(zson.QuotedName(e.Name))
+				c.write(":")
+				c.expr(e.Value, "")
+			case *dag.Spread:
+				c.write("...")
+				c.expr(e.Expr, "")
+			default:
+				c.write("zfmt: unknown record elem type: %T", e)
+			}
+		}
+		c.write("}")
+	case *dag.ArrayExpr:
+		c.write("[")
+		c.vectorElems(e.Elems)
+		c.write("]")
+	case *dag.SetExpr:
+		c.write("|[")
+		c.vectorElems(e.Elems)
+		c.write("]|")
+	case *dag.MapExpr:
+		c.write("|{")
+		for k, e := range e.Entries {
+			if k != 0 {
```

Comment:
Nit: I think the intention here is a little clearer with >.
```suggestion
			if k > 0 {
```